src/commands/gist.ts

```diff
@@ -376,6 +382,258 @@ export async function registerGistCommands(context: ExtensionContext) {
+                    const fileContent: Uint8Array = await workspace.fs.readFile(
+                      fileNameToUri(node!.gist.id, rawFileName)
+                    );
+
+                    downloadFile(newFileUri, fileContent);
+                  }
+                } catch (e) {
+                  // @todo: add Output trace to show the error
+                  output?.appendLine(
+                    `Error downloading gist: ${e}`,
+                    messageType.Error
+                  );
+                }
+              }
+            }
+          }
+        );
+      }
+    )
+  );
+
+  context.subscriptions.push(
+    commands.registerCommand(
+      `${EXTENSION_NAME}.downloadAllGists`,
+      async (targetNode?: any, multiSelectNodes?: any[]) => {
+        await ensureAuthenticated();
+
+        let folder = await window.showOpenDialog({
+          canSelectFiles: false,
+          canSelectFolders: true,
+          canSelectMany: false
+        });
+
+        let overwrite = new confirmOverwrite();
+        window.withProgress(
+          {
+            location: ProgressLocation.Notification,
+            title: "Downloading gists...",
+            cancellable: true
+          },
+          async (progress, token) => {
+            token.onCancellationRequested(() => {
+              output?.appendLine(
+                "Gist download cancelled by user.",
+                messageType.Info
+              );
+              overwrite.cancel();
+              return;
+            });
+
+            if (folder) {
+              for (const node of multiSelectNodes || [targetNode]) {
+                try {
+                  // create a folder for the user
+                  // gitGroupFolderName is the folder that will contain all gists downloaded for Gist group, e.g. followedUsers
+                  let gistGroupFolderName = "";
+                  let gists: any;
+
+                  [gistGroupFolderName, gists] = getGistFolderAndStore(node);
+
+                  let userFolderName =
+                    ensureIsValidFileSystemName(gistGroupFolderName);
+                  // create the folder for the Gist, use the Gist label as the name
+                  // @investigate: can treeNode.label be undefined?
+                  let newFolder = Uri.joinPath(folder[0], userFolderName);
+
+                  // check if the user wants to overwrite the folder
+                  let canOverwrite = await overwrite.confirm(newFolder);
+
+                  if (
+                    (overwrite.userChoice as confirmOverwriteOptions) ===
+                    confirmOverwriteOptions.Cancel
+                  ) {
+                    // the user cancelled the operation, so we can just return
+                    return;
+                  }
+
+                  if (!canOverwrite) {
+                    continue;
+                  }
+
+                  // create the folder for the Gist
+                  workspace.fs.createDirectory(newFolder);
```

Comment:
@lostintangent I'm not super-happy with this chunk of code, I have to repeat it twice in this method and once more in `downloadGist`, but it works; if you have suggestions I'm happy to update it.
The main problem I had is that it is not possible to interrups a loop (e.g. `continue` on L549) from another method. I experimented a bit with a recursive method but that was also ugly and not working properly so I kept this implementation for now.